packages/ketcher-polymer-editor-react/src/components/monomerLibrary/RnaBuilder/RnaEditor/RnaEditorExpanded/RnaEditorExpanded.tsx

```diff
@@ -74,6 +76,26 @@ export const RnaEditorExpanded = ({
   ] as const;
 
   const activeMonomerGroup = useAppSelector(selectActiveRnaBuilderItem);
+  const activePresetFullName = selectPresetFullName(activePreset);
+  const [presetName, setPresetName] = useState('');
+  const [editedPresetName, setEditedPresetName] = useState(false);
+
+  useEffect(() => {
+    if (
+      activeMonomerGroup !== RnaBuilderPresetsItem.Presets &&
+      !editedPresetName
+    ) {
+      setPresetName(activePresetFullName);
+    }
+  }, [activeMonomerGroup, activePresetFullName, editedPresetName]);
+
+  onChangeName = (event) => {
```

Comment:
Maybe it is possible to specify type for event?

Reply:
good idea!